src/utils/fs.rs

```diff
@@ -80,6 +80,77 @@ pub fn nice_directory_display(os_str: impl AsRef<OsStr>) -> Cow<'static, str> {
     }
 }
 
+/// Try to detect the file extension by looking for known magic strings
+pub fn try_infer(path: &Path) -> Option<Extension> {
+    fn is_zip(buf: &[u8]) -> bool {
+        buf.len() > 3
+            && buf[0] == 0x50
```

Comment:
Where you got the information about these magic numbers from? Could you add a comment in code so that we can refer to it later? (or send it here and I'll make sure to add it later).

Reply:
https://en.wikipedia.org/wiki/List_of_file_signatures